pet-editor/pet-editor.js

```diff
@@ -33,11 +34,15 @@ petForm.addEventListener('submit', async (e) => {
     const imagePath = `pets/${randomFolder}/${imageFile.name}`;
     // > Part A: Call upload image with the bucket ("images"),
     // the imagePath, and the imageFile - and store the returned url
+    const url = await uploadImage('images', imagePath, imageFile);
 
     const pet = {
         // > Part B: add the name, bio, and image_url fields to the pet object
+        name: formData.get('name'),
+        bio: formData.get('bio'),
+        image_url: url,
     };
-
+    const response = await createPet(pet);
```

Comment:
You don't have a redirect set up so It hard for the user to know when they have uploaded or if it is slow or something is wrong.